clc/modules/object-storage/src/main/java/com/eucalyptus/objectstorage/ObjectStorageGateway.java

```diff
@@ -877,14 +898,23 @@ public PostObjectResponseType postObject(PostObjectType request) throws S3Except
    */
   @Override
   public DeleteObjectResponseType deleteObject(final DeleteObjectType request) throws S3Exception {
-    ObjectEntity objectEntity;
+    ObjectEntity objectEntity = null;
+    Bucket bucket = null;
     try {
+      //LPT Bug! Throws exception on anonymous access to an object
+      //bucket = getBucketAndCheckAuthorization(request);
+      //LPT The fix:
```

Comment:
Why skip authorization checks? Object deletion should check whether user has WRITE permission on bucket - getBucketAndCheckAuthorization() performs that very check. Commenting it out lets anyone delete objects from any bucket